pkg/tap/link.go

```diff
@@ -82,6 +83,9 @@ func (e *LinkEndpoint) MaxHeaderLength() uint16 {
 }
 
 func (e *LinkEndpoint) MTU() uint32 {
+	if e.mtu < 0 || e.mtu > math.MaxUint32 {
+		e.mtu = math.MaxUint32
+	}
 	return uint32(e.mtu)
```

Comment:
Imo it’s cleaner to change the type of `e.mtu` in `LinkEndpoint`, and then check for overflows when it’s initialized:
```
@@ -14,7 +15,7 @@ import (
 
 type LinkEndpoint struct {
        debug      bool
-       mtu        int
+       mtu        uint32
        mac        tcpip.LinkAddress
        ip         string
        virtualIPs map[string]struct{}
```